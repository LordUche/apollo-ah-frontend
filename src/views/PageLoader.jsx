/* eslint-disable react/destructuring-assignment */
import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@emotion/core';
// First way to import
import { ClipLoader } from 'react-spinners';
// Another way to impor

const override = css`
    display: block;
    margin: 0 auto;
    border-color: red;
`;

/**
 * @class PageLoader - The loader component
 * @extends {React.Component}
 */
class PageLoader extends React.Component {
  /**
   * @static {propTypes} - Defining the type for the loading props
   * @memberof PageLoader
   */
  static propTypes = {
    isLoading: PropTypes.bool,
  }

  /**
   * @static {defaultProps} - Defining the default type for the isLoading props
   * @memberof PageLoader
   */
  static defaultProps = {
    isLoading: false,
  }

  /**
   *Creates an instance of PageLoader.
   * @param {*} props - The props passed in to the component
   * @memberof PageLoader
   */
  constructor(props) {
    super(props);
    this.state = {
      loading: props.isLoading,
    };
  }

  /**
   * @returns {JSX} - The loading component
   * @memberof PageLoader
   */
  render() {
    return (
      <div className="sweet-loading">
        <ClipLoader
          css={override}
          sizeUnit="px"
          size={35}
          color="#8b01be"
          loading={this.state.loading}
        />
      </div>
    );
  }
}


export default PageLoader;
