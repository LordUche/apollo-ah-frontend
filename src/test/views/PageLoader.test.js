import React from 'react';
import { shallow } from 'enzyme';
import PageLoader from '../../views/PageLoader';

describe('<PageLoader />', () => {
  it('should render the PageLoader', () => {
    const wrapper = shallow(<PageLoader />);
    expect(wrapper).toMatchSnapshot();
  });
});
