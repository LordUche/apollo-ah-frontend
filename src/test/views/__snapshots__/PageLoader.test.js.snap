// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLoader /> should render the PagLoader 1`] = `
<div
  className="sweet-loading"
>
  <onlyUpdateForKeys(Loader)
    color="#8b01be"
    css={
      Object {
        "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlBhZ2VMb2FkZXIuanN4Il0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQVFvQiIsImZpbGUiOiJQYWdlTG9hZGVyLmpzeCIsInNvdXJjZXNDb250ZW50IjpbIi8qIGVzbGludC1kaXNhYmxlIHJlYWN0L2Rlc3RydWN0dXJpbmctYXNzaWdubWVudCAqL1xuaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCBQcm9wVHlwZXMgZnJvbSAncHJvcC10eXBlcyc7XG5pbXBvcnQgeyBjc3MgfSBmcm9tICdAZW1vdGlvbi9jb3JlJztcbi8vIEZpcnN0IHdheSB0byBpbXBvcnRcbmltcG9ydCB7IENsaXBMb2FkZXIgfSBmcm9tICdyZWFjdC1zcGlubmVycyc7XG4vLyBBbm90aGVyIHdheSB0byBpbXBvclxuXG5jb25zdCBvdmVycmlkZSA9IGNzc2BcbiAgICBkaXNwbGF5OiBibG9jaztcbiAgICBtYXJnaW46IDAgYXV0bztcbiAgICBib3JkZXItY29sb3I6IHJlZDtcbmA7XG5cbi8qKlxuICogQGNsYXNzIFBhZ2VMb2FkZXIgLSBUaGUgbG9hZGVyIGNvbXBvbmVudFxuICogQGV4dGVuZHMge1JlYWN0LkNvbXBvbmVudH1cbiAqL1xuY2xhc3MgUGFnZUxvYWRlciBleHRlbmRzIFJlYWN0LkNvbXBvbmVudCB7XG4gIC8qKlxuICAgKiBAc3RhdGljIHtwcm9wVHlwZXN9IC0gRGVmaW5pbmcgdGhlIHR5cGUgZm9yIHRoZSBsb2FkaW5nIHByb3BzXG4gICAqIEBtZW1iZXJvZiBQYWdlTG9hZGVyXG4gICAqL1xuICBzdGF0aWMgcHJvcFR5cGVzID0ge1xuICAgIGlzTG9hZGluZzogUHJvcFR5cGVzLmJvb2wsXG4gIH1cblxuICAvKipcbiAgICogQHN0YXRpYyB7ZGVmYXVsdFByb3BzfSAtIERlZmluaW5nIHRoZSBkZWZhdWx0IHR5cGUgZm9yIHRoZSBpc0xvYWRpbmcgcHJvcHNcbiAgICogQG1lbWJlcm9mIFBhZ2VMb2FkZXJcbiAgICovXG4gIHN0YXRpYyBkZWZhdWx0UHJvcHMgPSB7XG4gICAgaXNMb2FkaW5nOiBmYWxzZSxcbiAgfVxuXG4gIC8qKlxuICAgKkNyZWF0ZXMgYW4gaW5zdGFuY2Ugb2YgUGFnZUxvYWRlci5cbiAgICogQHBhcmFtIHsqfSBwcm9wcyAtIFRoZSBwcm9wcyBwYXNzZWQgaW4gdG8gdGhlIGNvbXBvbmVudFxuICAgKiBAbWVtYmVyb2YgUGFnZUxvYWRlclxuICAgKi9cbiAgY29uc3RydWN0b3IocHJvcHMpIHtcbiAgICBzdXBlcihwcm9wcyk7XG4gICAgdGhpcy5zdGF0ZSA9IHtcbiAgICAgIGxvYWRpbmc6IHByb3BzLmlzTG9hZGluZyxcbiAgICB9O1xuICB9XG5cbiAgLyoqXG4gICAqIEByZXR1cm5zIHtKU1h9IC0gVGhlIGxvYWRpbmcgY29tcG9uZW50XG4gICAqIEBtZW1iZXJvZiBQYWdlTG9hZGVyXG4gICAqL1xuICByZW5kZXIoKSB7XG4gICAgcmV0dXJuIChcbiAgICAgIDxkaXYgY2xhc3NOYW1lPVwic3dlZXQtbG9hZGluZ1wiPlxuICAgICAgICA8Q2xpcExvYWRlclxuICAgICAgICAgIGNzcz17b3ZlcnJpZGV9XG4gICAgICAgICAgc2l6ZVVuaXQ9XCJweFwiXG4gICAgICAgICAgc2l6ZT17MzV9XG4gICAgICAgICAgY29sb3I9XCIjOGIwMWJlXCJcbiAgICAgICAgICBsb2FkaW5nPXt0aGlzLnN0YXRlLmxvYWRpbmd9XG4gICAgICAgIC8+XG4gICAgICA8L2Rpdj5cbiAgICApO1xuICB9XG59XG5cblxuZXhwb3J0IGRlZmF1bHQgUGFnZUxvYWRlcjtcbiJdfQ== */",
        "name": "3vn6pu-override",
        "styles": "display:block;margin:0 auto;border-color:red;label:override;",
      }
    }
    loading={false}
    size={35}
    sizeUnit="px"
  />
</div>
`;

exports[`<PageLoader /> should render the PageLoader 1`] = `
<div
  className="sweet-loading"
>
  <onlyUpdateForKeys(Loader)
    color="#8b01be"
    css={
      Object {
        "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlBhZ2VMb2FkZXIuanN4Il0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQVFvQiIsImZpbGUiOiJQYWdlTG9hZGVyLmpzeCIsInNvdXJjZXNDb250ZW50IjpbIi8qIGVzbGludC1kaXNhYmxlIHJlYWN0L2Rlc3RydWN0dXJpbmctYXNzaWdubWVudCAqL1xuaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCBQcm9wVHlwZXMgZnJvbSAncHJvcC10eXBlcyc7XG5pbXBvcnQgeyBjc3MgfSBmcm9tICdAZW1vdGlvbi9jb3JlJztcbi8vIEZpcnN0IHdheSB0byBpbXBvcnRcbmltcG9ydCB7IENsaXBMb2FkZXIgfSBmcm9tICdyZWFjdC1zcGlubmVycyc7XG4vLyBBbm90aGVyIHdheSB0byBpbXBvclxuXG5jb25zdCBvdmVycmlkZSA9IGNzc2BcbiAgICBkaXNwbGF5OiBibG9jaztcbiAgICBtYXJnaW46IDAgYXV0bztcbiAgICBib3JkZXItY29sb3I6IHJlZDtcbmA7XG5cbi8qKlxuICogQGNsYXNzIFBhZ2VMb2FkZXIgLSBUaGUgbG9hZGVyIGNvbXBvbmVudFxuICogQGV4dGVuZHMge1JlYWN0LkNvbXBvbmVudH1cbiAqL1xuY2xhc3MgUGFnZUxvYWRlciBleHRlbmRzIFJlYWN0LkNvbXBvbmVudCB7XG4gIC8qKlxuICAgKiBAc3RhdGljIHtwcm9wVHlwZXN9IC0gRGVmaW5pbmcgdGhlIHR5cGUgZm9yIHRoZSBsb2FkaW5nIHByb3BzXG4gICAqIEBtZW1iZXJvZiBQYWdlTG9hZGVyXG4gICAqL1xuICBzdGF0aWMgcHJvcFR5cGVzID0ge1xuICAgIGlzTG9hZGluZzogUHJvcFR5cGVzLmJvb2wsXG4gIH1cblxuICAvKipcbiAgICogQHN0YXRpYyB7ZGVmYXVsdFByb3BzfSAtIERlZmluaW5nIHRoZSBkZWZhdWx0IHR5cGUgZm9yIHRoZSBpc0xvYWRpbmcgcHJvcHNcbiAgICogQG1lbWJlcm9mIFBhZ2VMb2FkZXJcbiAgICovXG4gIHN0YXRpYyBkZWZhdWx0UHJvcHMgPSB7XG4gICAgaXNMb2FkaW5nOiBmYWxzZSxcbiAgfVxuXG4gIC8qKlxuICAgKkNyZWF0ZXMgYW4gaW5zdGFuY2Ugb2YgUGFnZUxvYWRlci5cbiAgICogQHBhcmFtIHsqfSBwcm9wcyAtIFRoZSBwcm9wcyBwYXNzZWQgaW4gdG8gdGhlIGNvbXBvbmVudFxuICAgKiBAbWVtYmVyb2YgUGFnZUxvYWRlclxuICAgKi9cbiAgY29uc3RydWN0b3IocHJvcHMpIHtcbiAgICBzdXBlcihwcm9wcyk7XG4gICAgdGhpcy5zdGF0ZSA9IHtcbiAgICAgIGxvYWRpbmc6IHByb3BzLmlzTG9hZGluZyxcbiAgICB9O1xuICB9XG5cbiAgLyoqXG4gICAqIEByZXR1cm5zIHtKU1h9IC0gVGhlIGxvYWRpbmcgY29tcG9uZW50XG4gICAqIEBtZW1iZXJvZiBQYWdlTG9hZGVyXG4gICAqL1xuICByZW5kZXIoKSB7XG4gICAgcmV0dXJuIChcbiAgICAgIDxkaXYgY2xhc3NOYW1lPVwic3dlZXQtbG9hZGluZ1wiPlxuICAgICAgICA8Q2xpcExvYWRlclxuICAgICAgICAgIGNzcz17b3ZlcnJpZGV9XG4gICAgICAgICAgc2l6ZVVuaXQ9XCJweFwiXG4gICAgICAgICAgc2l6ZT17MzV9XG4gICAgICAgICAgY29sb3I9XCIjOGIwMWJlXCJcbiAgICAgICAgICBsb2FkaW5nPXt0aGlzLnN0YXRlLmxvYWRpbmd9XG4gICAgICAgIC8+XG4gICAgICA8L2Rpdj5cbiAgICApO1xuICB9XG59XG5cblxuZXhwb3J0IGRlZmF1bHQgUGFnZUxvYWRlcjtcbiJdfQ== */",
        "name": "3vn6pu-override",
        "styles": "display:block;margin:0 auto;border-color:red;label:override;",
      }
    }
    loading={false}
    size={35}
    sizeUnit="px"
  />
</div>
`;
